ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -522,9 +523,14 @@ export default class ConfirmTransactionBase extends Component {
         </div>
       ) : null;
 
-    const { simulationData } = txData;
+    const { simulationData, approvalTxId } = txData;
 
-    const simulationDetails = (
+    // if transaction reverts in simulation and there's an approval transaction, we don't want to show the simulation details
+    const isUnsimulable =
+      approvalTxId &&
+      simulationData?.error.code === SimulationErrorCode.Reverted;
+
+    const simulationDetails = isUnsimulable ? undefined : (
```

Comment:
Given the complexity and many overlapping scenarios, the `SimulationDetails` component generates its own error messages based on the result.

Why is this a special case in which the error is not shown? Is it desirable to introduce that inconsistency?

If we definitely want this, could we instead provide a new property such as `hideErrors` so we can keep the error message logic defined in one place within the component?